### DISCUSSION TITLE
How can i extract the feedback of "good response" or "bad response"?

### DISCUSSION THREAD

**Discussion by cesarandreslopez** [open-webui/open-webui#1836] [General]
**Is your feature request related to a problem? Please describe.**
It would be great if I could dump all the feedback provided by a user.

**Describe the solution you'd like**
Ideally an endpoint where I could download all the "good response" or "bad response" items that users has given

**Describe alternatives you've considered**
Couldn't really understand where is the feedback for a particular model stored, so I don't know how to extract it. 

**Additional context**
This could be a great feature to improve models using open-webui

**Comment by flefevre** [comment]
at first, it seems you could use the fonction "export all chats" by clicking in the web ui interface.
but i do agree, there is a lack of documentation of this feature.

it will be good also if this information is push also into the callback feature linked to langfuse.

**Reply by abhishek-ch** [reply]
I don't find this option, just got export chats that  has no feedback info

**Reply by latent-variable** [reply]
I am working on doing this too, if you have any updates on the ability to extracted the feedback that would be very much appreciated

**Comment by aleixdorca** [comment]
I'm sure this is not the answer you are looking for because it would be great to have this infor straight in the UI.

Right now the values of "Good" and "Bad" answers are stored in the DB as annotations to the chat.

You can download the whole database from Admin Panel -> Settings -> Database -> Download database.

It is a SQLite 3 DB and you can query the chat table like so:

```
$ sqlite3 webui.db
sqlite> select * from chat where chat LIKE '%"annotations"%';
```

This displays all chats and their contents that have "annotations" (if no rating has been provided this is not present in the chat JSON structure).

You could parse the JSON structure and evaluate the responses yourself. Again, not what you are looking for, maybe.

Cheers.

**Reply by denzalman** [reply]
is it somehow possible to get these annotations from inlet() or outlet() methods? it could be great to work with it inside pipeline flows.

**Reply by cspengler-plexxis** [reply]
there's an endpoint to retrieve the DB/annotations: api/v1/chats/all/db

I have not done it myself but perhaps you could make a request to that within your inlet or outlet (in the header you'll have to include the JWT token for your account, you can find it under Settings -> Account), and go from there.

**Reply by Devin-M-Smith** [reply]
what does the ui do with these annotations? Is it plugged into the context of all chats? I _assume_ and hope that the data doesn't leave my llm environment.

In short: Does this actually improve my llm responses?

**Reply by cspengler-plexxis** [reply]
My understanding is that the annotations are not automatically plugged into the chats at all. They are simply stored (locally) in JSON for admins to retrieve and use however they choose. So giving feedback does not automatically improve LLM responses, nor does any of it leave your environment.

**Reply by Devin-M-Smith** [reply]
> My understanding is that the annotations are not automatically plugged into the chats at all. They are simply stored (locally) in JSON for admins to retrieve and use however they choose. So giving feedback does not automatically improve LLM responses, nor does any of it leave your environment.

Thanks for the info. I'll make use of it.

**Reply by Lixiyao-meow** [reply]
Thank you for this answer. Although they might have changed the field name, it is "annotation" now. So the query should be
`select * from chat where chat LIKE '%"annotation"%';`

**Comment by verynewtocode** [comment]
do openweb ui auto build on feedback? if i write a comment, i dont think it will contribute
